src/applications/check-in/components/AppointmentDisplay/AppointmentActionVaos.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { parseISO } from 'date-fns';
+// eslint-disable-next-line import/no-unresolved
+import { recordEvent } from '@department-of-veterans-affairs/platform-monitoring/exports';
+import { api } from '../../api';
+
+import { createAnalyticsSlug } from '../../utils/analytics';
+import { useFormRouting } from '../../hooks/useFormRouting';
+import { ELIGIBILITY, areEqual } from '../../utils/appointment/eligibility';
+
+import { appointmentWasCheckedInto } from '../../actions/day-of';
+
+import { CheckInButton } from './CheckInButton';
+import { useUpdateError } from '../../hooks/useUpdateError';
+
+const AppointmentActionVaos = props => {
+  const { appointment, router, token, event } = props;
+
+  const dispatch = useDispatch();
+  const setSelectedAppointment = useCallback(
+    appt => {
+      dispatch(appointmentWasCheckedInto(appt));
+    },
+    [dispatch],
+  );
+  const { updateError } = useUpdateError();
+
+  const { jumpToPage } = useFormRouting(router);
+  const onClick = useCallback(
```

Comment:
Could we move the onclick and the dispatch to the AppointmentBlockVaos and pass the onclick through as a prop? That would probably remove the need for this file. I don't mind having all the code in the parent component. That way we could avoid having to pass the router down or the component as a prop.

Reply:
yeah that makes sense to me!